addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/discovery/HomematicDeviceDiscoveryService.java

```diff
@@ -59,51 +65,132 @@ public void deactivate() {
     @Override
     protected void startScan() {
         logger.debug("Starting Homematic discovery scan");
+        enableInstallMode();
         loadDevices();
     }
 
+    /**
+     * Will set controller in <i>installMode==true</i>, but only if the bridge
+     * is ONLINE (e.g. not during INITIALIZATION).
+     */
+    private void enableInstallMode() {
+        try {
+            HomematicGateway gateway = bridgeHandler.getGateway();
+            ThingStatus bridgeStatus = null;
+            
+            if (bridgeHandler.getThing() != null) {
+                Thing bridge = bridgeHandler.getThing();
+                bridgeStatus = bridge.getStatus();
+                updateInstallModeDuration(bridge);
+            }
+            if (ThingStatus.ONLINE == bridgeStatus) {
+                gateway.setInstallMode(true, installModeDuration);
+
+                int remaining = gateway.getInstallMode();
+                if (remaining > 0) {
+                    setIsInInstallMode();
+                    logger.debug("Successfully put controller in install mode. Remaining time: {} seconds", remaining);
+                } else {
+                    logger.error("Controller did not accept requested install mode");
```

Comment:
Just to be sure: We can only end up here if there is something seriously wrong (i.e. it can be considered a software bug) and it is a non-recoverable situation, correct? If so, a stacktrace could be added to make it easier to track down the situation.

Reply:
It is very unlikely to happen (if the datapoint is available, the command should be handled without problems), but it is not a non-recoverable situation. The user requested install mode but this was not achieved. However, he may simply retry successfully. Therefore I will change it to warning here.